### PR TITLE
Scaffold: Artists

### PR DESCRIPTION
**Pull request by varffs**
- Language might want checking
- Missing the associated artworks
- Missing the youtube/embed field

(Do we want to use english for database value names and spanish for public facing? Is that ok or should we make everything spanish?)

### CODE REVIEW

File: src/components/artistas/ArtistaForm.jsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+  state = {
+    active: '',
+    name: '',
+    country: '',
+    gallery: '',
+    galleryUrl: '',
+    bioEditorState: '',
+    bioRawContent: '',
+    active: false,
```
Comment (mixcas) [inline src/components/artistas/ArtistaForm.jsx]: active is repeated here

File: src/components/artistas/ArtistaForm.jsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    error: {
+      message: '',
+    },
+    isLoading: false,
+  }
+
+  constructor(props) {
+    super(props);
+
+    // If component recieves noticia as prop we merge it with initial state (used for editing)
```
Comment (mixcas) [inline src/components/artistas/ArtistaForm.jsx]: this comment says `noticia` still

File: src/components/artistas/ArtistaForm.jsx [hunk 3]
```diff
@@ -0,0 +1,221 @@
+import React, { Component } from 'react';
+import { firebaseConnect } from 'react-redux-firebase';
+import { withRouter } from 'react-router-dom';
+
+import { convertFromRaw, convertToRaw, EditorState } from 'draft-js';
+import { Editor } from 'react-draft-wysiwyg';
+
+import 'react-datepicker/dist/react-datepicker.css';
```
Comment (mixcas) [inline src/components/artistas/ArtistaForm.jsx]: this are only used for the datepicker and are not necessary here

File: src/components/artistas/ArtistaForm.jsx [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+    this.setState({ isLoading: true })
+
+    this.props.firebase
+      .push('artistas', {
+        name,
+        active,
+        country,
+        gallery,
+        galleryUrl,
+        bioRawContent
```
Comment (mixcas) [inline src/components/artistas/ArtistaForm.jsx]: missing trailing comma here

File: src/components/artistas/ArtistaForm.jsx [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+    });
+  }
+
+  render() {
+    return (
+      <form onSubmit={event => event.preventDefault()}>
+        <div className='grid-row margin-bottom-basic justify-end'>
+          <div className='grid-item'>
+            <button className='button' onClick={() => this.props.id ? this.updateArtista() : this.addArtista()}>
+              Guardar{ this.props.id ? '' : ' Nueva'}
```
Comment (mixcas) [inline src/components/artistas/ArtistaForm.jsx]: should say "Nuevo"

File: src/components/artistas/ArtistaForm.jsx [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+          </div>
+        </div>
+
+        <div className='grid-row margin-bottom-basic'>
+          <div className='grid-item item-s-12'>
+            <h4 className='font-size-small font-bold margin-bottom-tiny'><label htmlFor='name'>Nombre</label></h4>
+            <input
+              id='name'
+              name='name'
+              ref={ ref => this.name = ref }
```
Comment (mixcas) [inline src/components/artistas/ArtistaForm.jsx]: Ref is not being used anywhere

File: src/components/artistas/ArtistaForm.jsx [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+        </div>
+
+        <div className='grid-row margin-bottom-basic'>
+          <div className='grid-item'>
+            <h4 className='font-size-small font-bold margin-bottom-tiny'><label htmlFor='active'>Activo</label></h4>
+            <div className='grid-row align-items-center'>
+              <input
+                id='active'
+                name='active'
+                ref={ ref => this.published = ref }
```
Comment (mixcas) [inline src/components/artistas/ArtistaForm.jsx]: If this refs come from the noticias code they are wrong in both places then

File: src/components/artistas/ArtistaForm.jsx [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+              disabled={this.state.isLoading}
+              value={this.state.gallery}
+              onChange={ event => this.setState({ gallery: event.target.value })}
+            />
+          </div>
+        </div>
+
+        <div className='grid-row margin-bottom-basic'>
+          <div className='grid-item item-s-12'>
+            <h4 className='font-size-small font-bold margin-bottom-tiny'><label htmlFor='galleryUrl'>Galleria URL</label></h4>
```
Comment (mixcas) [inline src/components/artistas/ArtistaForm.jsx]: Should say 'Pagina web de galeria'. Sidenote: In spanish "ll" sounds like Y in "Yoyo".

File: src/containers/artistas/UpdateArtista.jsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+import { firebaseConnect, isLoaded, isEmpty } from 'react-redux-firebase';
+
+import ArtistaForm from '../../components/artistas/ArtistaForm.jsx';
+
+const UpdateArtista = ({ firebase, artista, id }) => {
+
+  if (!isLoaded(artista)) { // If not loaded…
+    return 'Loading'; // …show 'loading'
+  } else if (isEmpty(artista)) { // …else. If is empty…
+    return 'Error'; // …show 'Error?'
```
Comment (mixcas) [inline src/containers/artistas/UpdateArtista.jsx:13]: Here we should say "No se encuentró el Artista". Maybe improve this view with a link back to the ArtistsList?

File: src/containers/artistas/UpdateArtista.jsx [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+        </header>
+
+        <ArtistaForm id={id} artista={artista} />
+      </section>
+    );
+  }
+};
+
+export default compose(
+  withRouter,
```
Comment (mixcas) [inline src/containers/artistas/UpdateArtista.jsx]: @varffs can u test if this is needed here? Remove it and try going to the artist list, then to edit an artist then hit the back button and the link in the nav. try having another tab open on this view and check if the data updates properly while editing on another tab.

**Review by mixcas — CHANGES_REQUESTED** [review]
The code works fine. It just havs lil details that need fixing. If i have time today I will fix it, otherwise please do it yourself @varffs

File: src/components/artistas/ArtistaForm.jsx (hunk 4 above)
Comment (varffs) [reply src/components/artistas/ArtistaForm.jsx]: if that is the case we need linting?

File: src/components/artistas/ArtistaForm.jsx (hunk 6 above)
Comment (varffs) [reply src/components/artistas/ArtistaForm.jsx]: same as noticias tho

File: src/containers/artistas/UpdateArtista.jsx (hunk 9 above)
Comment (varffs) [reply src/containers/artistas/UpdateArtista.jsx:13]: Nice error handling for the admin is def desirable not essential thus could be a thing at the end

File: src/components/artistas/ArtistaForm.jsx (hunk 4 above)
Comment (mixcas) [reply src/components/artistas/ArtistaForm.jsx]: i think we can configure the linting rules that come with create-react-app

File: src/components/artistas/ArtistaForm.jsx (hunk 6 above)
Comment (mixcas) [reply src/components/artistas/ArtistaForm.jsx]: i imagined, lets fix those too

File: src/containers/artistas/UpdateArtista.jsx (hunk 9 above)
Comment (mixcas) [reply src/containers/artistas/UpdateArtista.jsx:13]: kk lets just use a more descriptive copy

**Comment by varffs** [comment]
I think this is ready now

**Pull request merged**